test/junit/scala/tools/nsc/reporters/ConsoleReporterTest.scala

```diff
@@ -0,0 +1,211 @@
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+import scala.reflect.internal.util._
+
+
+@RunWith(classOf[JUnit4])
+class ConsoleReporterTest {
+  val noPos = NoPosition
+  val source = "Test_ConsoleReporter"
+  val batchFile = new BatchSourceFile(source, "For testing".toList)
+  val posWithSource = new OffsetPosition(batchFile, 4)
+  val content = posWithSource.lineContent
+
+  def createConsoleReporter(inputForReader: String, writerOut: ByteArrayOutputStream, echoWriterOut: ByteArrayOutputStream = null): ConsoleReporter = {
+      val writer = new PrintWriter(writerOut)
+      val reader = new BufferedReader(new StringReader(inputForReader))
+      
+      /** create reporter with the same writer and echoWriter */
+      if(echoWriterOut == null)
+        new ConsoleReporter(new Settings(), reader, writer) 
+      else
+      {
+        val echoWriter = new PrintWriter(echoWriterOut)
+        new ConsoleReporter(new Settings(), reader, writer, echoWriter)
+      }
+  }
+
+  /** Create a reporter */
+  val writerOut = new ByteArrayOutputStream()
+  val echoWriterOut = new ByteArrayOutputStream()
+  val reporter = createConsoleReporter("s", writerOut, echoWriterOut)
+
+  /** Change maxerrs and maxwarns */
+  reporter.settings.maxerrs.value = 50
+  reporter.settings.maxwarns.value = 25
+
+
+  def testHelper(checkBlank: Boolean, hasPos: Boolean, message: String, severity: String): Unit = {
+    if(checkBlank) assertEquals("", writerOut.toString)
+    else {
+      if(!hasPos) assertEquals(severity + message, writerOut.toString.lines.next)
+      else {
+        val it = writerOut.toString.lines
+        assertEquals(source + ":1: " + severity + message, it.next)
+        assertEquals(content, it.next)
+        assertEquals("    ^", it.next)
+      }
+    }
+    writerOut.reset
+  }
+
+  
+  @Test
+  def printMessageTest(): Unit = {
+    reporter.printMessage("Hello World!")
+    testHelper(false, false, "Hello World!", "")
+
+    /** printMessage test with NoPosition */
+    reporter.printMessage(noPos, "Testing with NoPosition")
+    testHelper(false, false, "Testing with NoPosition", "")
+
+    /** printMessage test with DefinedPosition */
+    reporter.printMessage(posWithSource, "Testing with Defined Position")
+    testHelper(false, true, "Testing with Defined Position", "")
+  }
+
+
+  @Test
+  def echoTest(): Unit = {
+    reporter.echo("Hello World!")
+    assertEquals("Hello World!", echoWriterOut.toString.lines.next)
+
+    /** checking with constructor which the same writer and echoWriter */
+    val reporter2 = createConsoleReporter("r", writerOut)
+    reporter2.echo("Hello World!")
+    testHelper(false, false, "Hello World!", "")
+  }
+
+
+  @Test
+  def printTest(): Unit = {
+    /** print test with NoPosition */
+    reporter.print(noPos, "test", reporter.INFO)
+    testHelper(false, false, "test", "")
+
+    reporter.print(noPos, "test", reporter.WARNING)
+    testHelper(false, false, "test", "warning: ")
```

Comment:
The `false, false`, which I've forgotten what it means, is noise.

Sometimes a little table of truth values in code is tidy, but not as args. It can help to name them, `f(hasPos = false, isBlank = false, x, y)`, and in non-test code the repo requires named boolean args for this reason.

Eliminating the noise would take this to another level.